src/sentry/workflow_engine/processors/action.py

```
@@ -115,9 +124,22 @@ def filter_recently_fired_workflow_actions(
     actions_without_statuses_ids = {action.id for action in actions_without_statuses}
     filtered_actions = actions.filter(id__in=actions_to_include | actions_without_statuses_ids)
 
+    result_list: list[tuple[Action, Workflow]] = []
+    for action in filtered_actions:
+        dcg_action = action.dataconditiongroupaction_set.first()
+        if dcg_action:
+            workflow = filtered_action_groups[dcg_action.condition_group]
+            result_list.append((action, workflow))
+        else:
+            logger.error(
+                "Action %s has no associated DataConditionGroupAction",
+                action.id,
+                extra={"action_id": action.id},
+            )
```

So this seems like an invariant we'd want to potentially uphold. Does it make sense to continue executing the rest of this function when this occurs, or should we raise an exception and bail due to the data inconsistency this likely signifies?

---

hmm, thinking about this. i know this is a hot path so whatever we end up selecting needs to be resilient